core-web/libs/sdk/react/src/lib/models/index.ts

```diff
@@ -87,3 +87,44 @@ export interface DotCMSPageContext {
     };
     isInsideEditor: boolean;
 }
+
+export interface DotCMSContentlet {
+    archived: boolean;
+    baseType: string;
+    deleted?: boolean;
+    binary?: string;
+    binaryContentAsset?: string;
+    binaryVersion?: string;
+    contentType: string;
+    file?: string;
+    folder: string;
+    hasLiveVersion?: boolean;
+    hasTitleImage: boolean;
+    host: string;
+    hostName: string;
+    identifier: string;
+    inode: string;
+    image?: string;
+    languageId: number;
+    language?: string;
+    live: boolean;
+    locked: boolean;
+    mimeType?: string;
+    modDate: string;
+    modUser: string;
+    modUserName: string;
+    owner: string;
+    sortOrder: number;
+    stInode: string;
+    title: string;
+    titleImage: string;
+    text?: string;
+    url: string;
+    working: boolean;
+    body?: string;
+    contentTypeIcon?: string;
+    variant?: string;
+    __icon__?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any; // This is a catch-all for any other custom properties that might be on the contentlet.
+}
```

Comment:
This is duplicated in Astro, Angular and here. We have to do something.

cc: @fmontes @KevinDavilaDotCMS @rjvelazco

Reply:
I proposed the creation of a types library some time ago

Reply:
We should consider the `@dotcms/types` lib for this/next iteration.

That would unify the types throughout the project, eliminate duplicate code/types, and make development easier

Reply:
I agree with the `@dotcms/types`. 
Now that we are adding more frameworks, it will become more difficult to add the types later

Reply:
Yeah... we need to bring this in, I feel we can start to use `@dotcms/types` in our SDK and later we can star using it in `core-web`